sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4381,15 +4386,11 @@ pi_result piEnqueueMemUnmap(pi_queue Queue, pi_mem MemObj, void *MappedPtr,
 
   ze_event_handle_t ZeEvent = nullptr;
   if (Event) {
-    auto Res = piEventCreate(Queue->Context, Event);
+    auto Res = createEventAndAssociateQueue(Queue, Event,
+                                            PI_COMMAND_TYPE_MEM_BUFFER_UNMAP,
+                                            ZeCommandList, &ZeEvent, false);
```

Comment:
Please move the queue lock made in line 4425 before this line and remove the last parameter.

Reply:
Done. Thanks.